internal/validator/validator_test.go

```diff
@@ -0,0 +1,256 @@
+			expectedError: "",
+		},
+		{
+			name:          "Invalid Type - Uppercase",
+			messageType:   "Feat",
+			expectedError: `invalid commit message casing, "Feat" should be "feat"`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageType(&tc.messageType)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageScope tests the checkMessageScope function.
+func TestCheckMessageScope(t *testing.T) {
+	testCases := []struct {
+		name          string
+		scope         string
+		expectedError string
+	}{
+		{
+			name:          "Valid Scope - Lowercase",
+			scope:         "user",
+			expectedError: "",
+		},
+		{
+			name:          "Valid Scope - Empty",
+			scope:         "",
+			expectedError: "",
+		},
+		{
+			name:  "Invalid Scope - Uppercase",
+			scope: "User",
+			expectedError: `invalid commit message scope casing, "User" ` +
+				`should be "user"`},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageScope(&tc.scope)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageSubject tests the checkMessageSubject function.
+func TestCheckMessageSubject(t *testing.T) {
+	testCases := []struct {
+		name          string
+		subject       string
+		expectedError string
+	}{
+		{
+			name:          "Valid Subject - Lowercase",
+			subject:       "add a new feature",
+			expectedError: "",
+		},
+		{
+			name:    "Invalid Subject - Uppercase",
+			subject: "Add a new feature",
+			expectedError: "commit message subject should be lowercased & " +
+				"not end with a period(.)",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageSubject(&tc.subject)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageLength tests the checkMessageLength function.
+func TestCheckMessageLength(t *testing.T) {
+	testCases := []struct {
+		name          string
+		message       string
+		expectedError string
+	}{
+		{
+			name:          "Valid Length",
+			message:       "feat: add a new feature",
+			expectedError: "",
+		},
+		{
+			name: "Invalid Length",
+			message: "Add a new feature that makes the application run " +
+				"faster and more efficiently",
+			expectedError: fmt.Sprintf(
```

Comment:
If the `expectedError` is of the `error` interface type then you can use the [`fmt.Errorf()`](https://pkg.go.dev/fmt#Errorf) function to define an error message as well.